src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMockFactory.java

```diff
@@ -30,22 +30,18 @@ private String getRandomName()
 	{
 		String firstName = FIRST_NAMES[random.nextInt(FIRST_NAMES.length)];
 		String lastName = LAST_NAMES[random.nextInt(LAST_NAMES.length)];
-		return firstName + " " + lastName;
+		return firstName + "_" + lastName + random.nextInt(9999);
 	}
 
 	/**
 	 * Gets a unique random name. Any name that is generated will never be generated again.
 	 *
 	 * @return A unique random name.
+	 * @throws RuntimeException if no unique name is found after 10 tries
 	 */
 	private String getUniqueRandomName()
 	{
-		if (usedNames.size() >= 100)
-		{
-			throw new RuntimeException("Out of names");
-		}
-
-		while (true)
+		for (int i = 0; i < 10; i++)
```

Comment:
Imposing a hard limit of 10 tries here is a terrible idea.

While the current limitation of 100 names is not the best solution either, it ensured a reliable and consistent name generation. By imposing this limit of 10 tries, you run into the chance of generating 10 names that already existed.
This will now throw a `RuntimeException`, making the unit test fail.

In other words: There is now a probality for randomly failing a unit test, which defeats the point behind a unit test. Never make your unit tests dependent on randomness being in your favour.

Reply:
You could change the name generation to simply use an counter which is added to each name, making any doubling impossible: `<RandomPlayerName>0, <OtherRandomPlayerName>1, ...`

This would also remove the need to check, whether a name was already generated.

But if you don't want consecutive numbers, you could also generate a (short) hash value based on the counter and add this to the player string (then the check for doubling names may be needed in rare circumstances).